engine/src/main/java/org/terasology/world/block/BlockRegions.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.block;
+
+import org.joml.RoundingMode;
+import org.joml.Vector3f;
+import org.joml.Vector3fc;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+
+public class BlockRegions {
+    private BlockRegions() {
+    }
+
+    /**
+     * Creates a new region spanning the smallest axis-aligned bounding box (AABB) containing both, min and max
+     *
+     * @return new block region
+     */
+    public static BlockRegion createFromMinAndMax(Vector3ic min, Vector3ic max) {
+        return new BlockRegion().setMin(min).setMax(max);
+    }
+
+    /**
+     * Creates a new region spanning the smallest axis-aligned bounding box (AABB) containing both, min and max
+     *
+     * @return new block region
+     */
+    public static BlockRegion createFromMinAndMax(int minX, int minY, int minZ, int maxX, int maxY, int maxZ) {
+        return new BlockRegion().setMin(minX, minY, minZ).setMax(maxX, maxY, maxZ);
+    }
+
+    /**
+     * Creates a new region centered around {@code center} extending each side by {@code extents}. The resulting
+     * axis-aligned bounding box (AABB) will have a size of {@code 2 * extents}
+     *
+     * @return new block region
+     */
+    public static BlockRegion createFromCenterAndExtents(Vector3ic center, Vector3ic extents) {
+        return new BlockRegion().translate(center).addExtents(extents);
+    }
+
+    /**
+     * Creates a new region centered around {@code center} extending each side by {@code extents}. The computed min is
+     * rounded up (ceil), the computed max is rounded down (floor). Thus, the resulting axis-aligned bounding box (AABB)
+     * will only include integer points that are within the floating point area and have a size of {@code <= 2 *
+     * extents}
+     *
+     * @return new block region
+     */
+    public static BlockRegion createFromCenterAndExtents(Vector3fc center, Vector3fc extents) {
+        Vector3f min = center.sub(extents, new Vector3f());
+        Vector3f max = center.add(extents, new Vector3f());
+
+        return new BlockRegion().setMin(new Vector3i(min, RoundingMode.CEILING)).setMax(new Vector3i(max,
+                RoundingMode.FLOOR));
```

Comment:
for the min you also want to floor the value.

if your min is (.001,.001,.001) ceiling would be (1,1,1) which would be one block extra. .001 would be within (0,0,0)

Reply:
Hm, but doing a floor on the min value would cause the region to grow because of rounding. Consider the region `R` where `R_min = (-0.5, -0.5, -0.5)` and `R_max = (0.5, 0.5, 0.5)`. I'd think that `R.containsPoint(0, 0, 0)` is true, but false for all other points (with integer coordinates). 

If we would _floor_ the min values the region `R'` would span the AABB from (-1, -1, -1) to (0, 0, 0) and `R'.contains(-1, -1, -1)` would now hold. I think the caller should make sure that the rounded values are "correct", possibly by adding a (small) margin.